### PR TITLE
Add nested navigation

### PR DESCRIPTION
**Pull request by MancunianSam**
This uses the https://github.com/nationalarchives/tdr-components/ library which provides the javascript and css.

Most of the rest of the changes are removing the old code for the cache based solution.

### CODE REVIEW

File: app/controllers/AdditionalMetadataNavigationController.scala [hunk 1]
```diff
@@ -1,252 +1,35 @@
...
-      ids <- folderDescendantsCache.getDescendantFileIds(consignmentId, folderId, token)
-      _ = if (selectedFiles.containsSome(ids)) partSelectedCache.add(folderId)
-      descendantFilesSelected = selectedFiles.toSet.intersect(ids.toSet).size
-    } yield descendantFilesSelected
+  def submitFiles(consignmentId: UUID, metadataType: String): Action[AnyContent] = standardTypeAction(consignmentId) { implicit request: Request[AnyContent] =>
+    val fileIds = request.body.asFormUrlEncoded.getOrElse(Map()).keys.toList
+      .filter(_ != "csrfToken")
+      .map(UUID.fromString)
+    if(metadataType == "closure") {
+      Future(Redirect(routes.AddClosureMetadataController.addClosureMetadata(consignmentId, fileIds)))
```
Comment (TomJKing) [inline app/controllers/AdditionalMetadataNavigationController.scala:29]: This should be redirecting to the closure status page: `routes.AdditionalMetadataClosureStatusController.getClosureStatusPage`

**Review by TomJKing — COMMENTED** [review]
Looks good. Couple of minor comments. Thanks

File: app/services/ConsignmentService.scala [hunk 2]
```diff
@@ -134,6 +137,34 @@ class ConsignmentService @Inject()(val graphqlConfiguration: GraphQLConfiguratio
       .map(data => data.getConsignment.get)
   }
 
+  def getAllConsignmentFiles(consignmentId: UUID, token: BearerAccessToken): Future[File] = {
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply app/services/ConsignmentService.scala:140]: Good spot, I've removed it, the associated client and the `getAllDescendants` can be removed now as well as it was only called in `getConsignmentPaginatedFile`

File: app/controllers/AdditionalMetadataNavigationController.scala (hunk 1 above)
Comment (MancunianSam) [reply app/controllers/AdditionalMetadataNavigationController.scala:29]: I've changed it and updated the tests.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks.

**Pull request merged**